aw_core/config.py

```diff
@@ -1,12 +1,83 @@
 import os
 import logging
+from typing import Any, Dict, Union
 from configparser import ConfigParser
 
+from deprecation import deprecated
+import tomlkit
+
 from aw_core import dirs
+from aw_core.__about__ import __version__
 
 logger = logging.getLogger(__name__)
 
 
+def _merge(a: dict, b: dict, path=None):
+    """
+    Recursively merges b into a, with b taking precedence.
+
+    From: https://stackoverflow.com/a/7205107/965332
+    """
+    if path is None:
+        path = []
+    for key in b:
+        if key in a:
+            if isinstance(a[key], dict) and isinstance(b[key], dict):
+                _merge(a[key], b[key], path + [str(key)])
+            elif a[key] == b[key]:
+                pass  # same leaf value
+            else:
+                a[key] = b[key]
+        else:
+            a[key] = b[key]
+    return a
+
+
+def _comment_out_toml(s: str):
+    return "\n".join(["#" + line for line in s.split("\n")])
+
+
+def load_config_toml(
+    appname: str, default_config: str
+) -> Union[dict, tomlkit.container.Container]:
+    config_dir = dirs.get_config_dir(appname)
+    config_file_path = os.path.join(config_dir, "{}.toml".format(appname))
+
+    # Run early to ensure input is valid toml before writing
+    default_config_toml = tomlkit.parse(default_config)
+
+    # Override defaults from existing config file
+    if os.path.isfile(config_file_path):
+        with open(config_file_path, "r") as f:
+            config = f.read()
+        config_toml = tomlkit.parse(config)
+    else:
+        # TODO: If file doesn't exist, write with commented-out default config
+        with open(config_file_path, "w") as f:
+            f.write(_comment_out_toml(default_config))
```

Comment:
This is the same way that aw-server-rust does it.
Whether it's the best solution though I'm not sure as the default config might become outdated if it gets updated in the future. It's a decent solution anyway. The other possible solutions I have in mind are neither better or worse just different, so might as well just be consistent with how aw-server-rust does it.